include/pybind11/cast.h

```diff
@@ -34,6 +34,20 @@ template <typename Type> class opaque {
 
 NAMESPACE_BEGIN(detail)
 
+/// Convert from any type pointer to another avoding warnings about void* vs (void*()) incompatibility.
```

Comment:
Starting with GCC 4.8, this works I think: -Wno-error=pedantic :).
This could be a solution -- just ignore those warnings for pybind11 using the pragma push/pop feature for warnings.